lua/entities/gmod_wire_coldetector.lua

```diff
@@ -0,0 +1,250 @@
+	
+	timer.Simple(0, function()
+		if IsValid(ent) and (IsValid(data.HitEntity) or data.HitEntity:IsWorld()) and not IsValid(self.Stick) and  not IsValid(ent:GetParent()) then
+			local bonehit = 0
+			local boneour = 0
+			self.StickTo = data.HitEntity
+			
+			for i = 0, data.HitEntity:GetPhysicsObjectCount()-1 do
+				if data.HitEntity:GetPhysicsObjectNum(i) == data.HitObject then
+					bonehit = i
+					break
+				end
+			end
+			
+			for i = 0, ent:GetPhysicsObjectCount()-1 do
+				if ent:GetPhysicsObjectNum(i) == data.PhysObject then
+					boneour = i
+					break
+				end
+			end
+			
+			if IsValid(data.HitObject) then
+				data.PhysObject:SetPos(data.HitObject:LocalToWorld(poslocal))
+				if boneour == 0 then
+					ent:SetPos(data.HitObject:LocalToWorld(poslocal))
+				end
+			else
+				data.PhysObject:SetPos(data.HitEntity:LocalToWorld(poslocal))
+				if boneour == 0 then
+					ent:SetPos(data.HitEntity:LocalToWorld(poslocal))
+				end
+			end
+			
+			data.PhysObject:SetAngles(data.HitEntity:LocalToWorldAngles(anglocal))
+			
+			if boneour == 0 then
+				ent:SetAngles(data.HitEntity:LocalToWorldAngles(anglocal))
+			end
+				
+			if IsValid(data.HitObject) and (not self.NoPhysics or data.HitEntity:IsWorld()) then
+				self.Stick = constraint.Weld(ent, data.HitEntity, boneour, bonehit, 0, ent:GetPhysicsObjectCount() > 1)
+				if data.HitEntity:IsWorld() then
+					data.PhysObject:EnableMotion(false)
+					data.PhysObject:Sleep()
+				end
+			else
+				ent:PhysicsDestroy()
+				ent:SetSolid(SOLID_NONE)
+				ent:SetMoveType(MOVETYPE_NONE)
+				ent:SetParent(data.HitEntity)
+			end
+		end
+	end)
+end
+
+function ENT:RemoveLink()
+	self.StickTo = nil
+	if IsValid(self.Stick) then
+		self.Stick:Remove()
+	end
+	
+	if IsValid(self.LinkedEnt) then
+		if IsValid(self.LinkedEnt:GetParent()) then
+			self.LinkedEnt:SetParent(nil)
+			self.LinkedEnt:PhysicsInit(SOLID_VPHYSICS)
+			self.LinkedEnt:SetSolid(SOLID_VPHYSICS)
+			self.LinkedEnt:SetMoveType(MOVETYPE_VPHYSICS)
+			timer.Simple(0, function() 
+				if IsValid(self.LinkedEnt) then
+					self.LinkedEnt:GetPhysicsObject():Wake() 
+				end
+			end)
+		end
+	end
+	
+end
+
+function ENT:TriggerInput(name, value)
+	if name == "Stick to world" then
+		self.StickToWorld = value
+		if value == 0 and self.StickTo and self.StickTo:IsWorld() then
+			self:RemoveLink()
+		end
+	elseif name == "Stick to props" then
+		self.StickToProps = value
+		if value == 0 and self.StickTo and not self.StickTo:IsNPC() and not self.StickTo:IsPlayer() and not self.StickTo:IsWorld() then
+			self:RemoveLink()
+		end
+	elseif name == "Stick to npc/players" then
+		self.StickToNPC=value
+		if value == 0 and self.StickTo and (self.StickTo:IsNPC() or self.StickTo:IsPlayer()) then
+			self:RemoveLink()
+		end
+	end
+end
+
+function ENT:Think()
+	if self.Collided then
+		WireLib.TriggerOutput(self, "Collided", 0)
+		self.Collided = false
+		self:ShowOutput()
+	end
+end
```

Comment:
I believe you need to call `BaseClass.Think` here to properly update certain wire things